rfcs/RFC Feature - USD in the Scene Pipeline.md

```diff
@@ -0,0 +1,545 @@
+    *   We will use the official Pixar library, instead of building our own logic for parsing USD files: [https://github.com/PixarAnimationStudios/USD](https://github.com/PixarAnimationStudios/USD)
+*   Update the scene builder to load USD files during Create Jobs, so asset references can be examined and resolved.  
+    *   Emit references to resolvable USD files as source dependencies.
+    *   Fail or error in the job if the reference can't be resolved.
+        *   [Open question - failure versus error. This is a two way door and can be easily changed at any time.](https://wiki.agscollab.com/display/lmbr/USD+in+the+Scene+Pipeline#USDintheScenePipeline-Shouldwefailajob,oremitanerrorifareferencetoanotherUSDfilecan%27tberesolved?)
+    *   Job dependencies won't yet be needed, because the flattening process will not need to reference product assets from other USD files.
+*   Update the scene API to reference both Assimp and the USD library.
+*   Update the scene API to detect USD format files, and branch to specialized USD loading.
+    *   Make sure this allows for both a simple load that just scans for references, independent of the more advanced loading.
+*   Write a custom asset path resolver, that we pass to the USD library, so it can resolve paths.
+    *   See USD documentation on Asset Resolution:
+        *   [https://graphics.pixar.com/usd/release/glossary.html#usdglossary-assetresolution](https://graphics.pixar.com/usd/release/glossary.html#usdglossary-assetresolution)
+        *   [https://graphics.pixar.com/usd/release/api/ar\_page\_front.html](https://graphics.pixar.com/usd/release/api/ar_page_front.html)
+*   In process job for a USD file in the Scene Builder, flatten the USD file using the USD library.
+*   In process job for a USD file in the Scene Builder, convert the flattened USD file to an O3DE scene.
+
+Open Questions
+==============
+
+Should we move Assimp to a Gem instead of using O3DE 3rdParty system?
+---------------------------------------------------------------------
+
+Moving forward with USD via Assimp will require regularly pulling new versions of Assimp as new features come online (example: references). Going through the 3rdParty library system is slower as it requires its own approval process via [O3DE's 3rd Party Package repo](https://github.com/o3de/3p-package-source). The downside to moving Assimp into a gem directly, is that we'd reimplement what's already in 3rdParty, for example, [git pulling a particular Assimp commit](https://github.com/o3de/3p-package-source/blob/main/package-system/assimp/build_config.json#L2-L3), [building](https://github.com/o3de/3p-package-source/blob/main/package-system/assimp/build_assimp_windows.cmd), and [packaging](https://github.com/o3de/3p-package-source/blob/main/package-system/assimp/install_assimp_windows.json) the library. This would also increase build time when compiling the editor source; Assimp takes 2 minutes to download and compile on my local 16-Core 3.4GHz PC.
+
```

Comment:
I would actually suggest moving Assimp into a `fetchcontent` type library, if it only takes 2 minutes to compile.  There's nothing in its build scripts I see that would make it incompatible with this.

Reply:
Thank you! Tracking here: https://github.com/o3de/o3de/pull/18273